include/eve/module/complex/complex.hpp

```diff
@@ -144,5 +145,139 @@ namespace eve
     {
       return Z{-real(z), -imag(z)};
     }
+
+    template<like<complex> Z1, like<complex> Z2>
+    EVE_FORCEINLINE friend auto& operator-= (Z1& self, Z2 const& o) noexcept
+    {
+      real(self) -= real(o);
+      imag(self) -= imag(o);
+      return self;
+    }
+
+    EVE_FORCEINLINE friend auto& operator-=(like<complex> auto& self, callable_i_ const&) noexcept
+    {
+      imag(self)--;
+      return self;
+    }
+
+    template<typename Z>
+    EVE_FORCEINLINE friend auto& operator-=(like<complex> auto& self, Z o) noexcept
+    requires(like<Z,Type> || std::convertible_to<Z,Type>)
+    {
+      real(self) -= o;
+      return self;
+    }
+
+    template<like<complex> Z1, real_value Z2>
+    EVE_FORCEINLINE friend  auto operator-(Z1 const& x, Z2 const& y) noexcept
+    requires(requires(as_wide_as_t<Z1,Z2> t) { t -= y; })
+    {
+      as_wide_as_t<Z1,Z2> that(x);
+      return that -= y;
+    }
+
+    template<real_value Z1, like<complex> Z2>
+    EVE_FORCEINLINE friend  auto operator-(Z1 const& x, Z2 const& y) noexcept
+    requires(requires(as_wide_as_t<Z2,Z1> t) { t -= y; })
+    {
+      as_wide_as_t<Z2,Z1> that{-y};
+      return x + that;
+    }
+
+    //==============================================================================================
+    // Operator *
+    //==============================================================================================
+    template<like<complex> Z1, like<complex> Z2>
+    EVE_FORCEINLINE friend auto& operator*= (Z1& self, Z2 const& o) noexcept
+    {
+      auto [a, b] = self;
+      auto [c, d] = o;
+      real(self) = diff_of_prod(a, c, b, d);
+      imag(self) = sum_of_prod(a, d, b, c);
+      return self;
+    }
+
+    EVE_FORCEINLINE friend auto& operator*=(like<complex> auto& self, callable_i_ const&) noexcept
+    {
+      auto[r,i] = self;
+      real(self) = -i;
+      imag(self) =  r;
+      return self;
+    }
+
+    template<typename Z>
+    EVE_FORCEINLINE friend auto& operator*=(like<complex> auto& self, Z o) noexcept
+    requires(like<Z,Type> || std::convertible_to<Z,Type>)
```

Comment:
Have a think about this one? I think this restriction is a bit weird.

Reply:
it means that we accept anything that is either the type or a scalar convertible to it as per the schema we drafted on which scalar/wide/complex mix we want.

That was discussed and vetted last PR.

Reply:
Discussed ot be fixed later